apps/print/backend/src/app.ts

```diff
@@ -216,37 +226,38 @@ export function buildApi(ctx: AppContext) {
         partyId: input.partyId,
       });
 
-      // TODO(nikhil): Actually print ballot, increment print count within transaction
+      const ballot = store.getBallot({
+        ballotStyleId,
+        precinctId: input.precinctId,
+        ballotType: input.ballotType,
+      });
+      if (!ballot || !ballot.encodedBallot) {
+        await logger.logAsCurrentRole(LogEventId.PrinterPrintRequest, {
+          message: 'No ballot found',
+          ballotProps: JSON.stringify({
+            precinctId: input.precinctId,
+            splitId: input.splitId,
+            partyId: input.partyId,
+            languageCode: input.languageCode,
+            ballotType: input.ballotType,
+          }),
+          disposition: 'failure',
+        });
+        return;
+      }
+
+      await printer.print({
+        data: Buffer.from(ballot.encodedBallot, 'base64'),
+        copies: input.copies,
+      });
```

Comment:
I wonder if in a future version we need to read the number of copies printed and update by that amount? If the print fails halfway through due to lack of paper or something else, this number will be incorrect.

Reply:
That's a good point, will note that for discussion

Reply:
It's a problem that has come up in Warren in v2 - they print ballots but due to some malfunction, some of the ballots don't print or are invalid - and their counts end up wrong. The most common cause - lack of paper - doesn't usually mess up the count if paper is then inserted. But there are other things that might!

I believe the eng lift is quite high. The CUPS server, which is easy to communicate with, doesn't really track job status. We'd have to make IPP requests directly to the printer and track some kind of ID for each job.

I think doing that would be cool and tighten up the product, but I'm not sure at what level of product maturity we should actually tackle it.